### PR TITLE
Adding custom retry amount to Machine.run()

### PR DESCRIPTION
**Pull request by gsegatti**
I've recently noted the retry logic for `run()` within `machine.py` has a fixed value of 30 retries and a 1 second sleep between each iteration.

I'd like to propose the possibility of letting the user set this amount via the `Machine` constructor. This way an user can await for a lesser time span between trying to run a `Machine` instance.

Please let me know if this would be something that makes sense as my last PR was quite a few months ago.

### CODE REVIEW

File: mkosi/machine.py [hunk 1]
```diff
@@ -48,9 +48,10 @@ def flush(self) -> None:
 
 
 class Machine:
-    def __init__(self, args: Sequence[str] = []) -> None:
+    def __init__(self, args: Sequence[str] = [], retry_amount: int = 30) -> None:
```
Comment (behrmann) [inline mkosi/machine.py:51]:
```suggestion
    def __init__(self, args: Sequence[str] = [], *, retry_amount: int = 30) -> None:
```
I'd prefer to make keyword arguments with defaults to only be callable via keyword

**Review by behrmann — COMMENTED** [review]
LGTM, with one nit. Would be nice to plumb this through to pytest in the machine test.

File: mkosi/machine.py (hunk 1 above)
Comment (DaanDeMeyer) [reply mkosi/machine.py:51]: Let's rename it to `num_retries` as well

Comment (gsegatti) [reply mkosi/machine.py]: Yep, via keyword seems better. Regarding your other comment, I didn't quite understand what "plumb through" the `num_retries` to pytest would be, so I tried to implement what I thought correct.

Let me know if I got it wrong.

Comment (behrmann) [reply mkosi/machine.py]: Sorry for being unclear: What I meant is to be able to run pytest such, that i can set `num_retries` on the CLI.

**Review by DaanDeMeyer — APPROVED** [review]

**Pull request merged**